lib/dry_twitter/sign_in/set_cookie.rb

```diff
@@ -0,0 +1,16 @@
+require "dry/transaction/operation"
+
+module DryTwitter
+  module SignIn
+    class SetCookie
+      include Dry::Transaction::Operation
+
+      def call(input)
+        env = input[:env]
+        env['rack.session'][:user_name] = input["user"]["user_name"]
```

Comment:
Do you guarantee uniques of username? if you want to use this you should update the validation rule.

Reply:
I am also not sure why you store this into the `env`? It would have made sens to store into `session`, can you explain your decision?

Reply:
I don't understand what do you mean.

Reply:
I dropped the operation and set user_name on session on route, the same as in Berg.